lib/sejfguru/assets/asset_importer.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Sejfguru.Assets.AssetImporter do
+
+  def import do
+    import_page(1)
+  end
+
```

Comment:
At first glance, it was really hard to see how and when do we stop fetching API.
Have you tried to handle errors using `with` statement?

Reply:
There is a while loop done (using recursion) that checks if response from Freshservice API has any assets at all. If not we exceeded the page number and we stop fetching another one. If any record is not saved (`{:error, _}`) we also stop fetching more pages.
I can try to refactor it, but it would be really nice if you could give me an idea how to write it so it's more readable :)

Maybe something like this looks better?
```elixir
def import do
   import_page(1)
end

defp import_page(page) do
   case import_assets(page) do
      [] -> { :ok, "Freshservice assets imported successfully from #{page-1} pages" }
      {:ok} -> import_page(page+1)
      {:error} -> { :error, "Freshservice assets import finished with error on page #{page}" }
   end
end

defp import_assets(page) do
   case fetch_assets(page) do
      [] -> []
      assets -> if Enum.all?(assets, fn(asset) -> insert_asset(asset) end), do: {:ok}, else: {:error}
   end
end
```
There are no errors raised during asset upsert, no validations are present. There is an `insert!` that will raise an error, but I'm not really sure if that will be of any help here. What do you think?